### PR TITLE
Get function offset instruction

### PR DESCRIPTION
**Pull request by dwightguth**
Adds an instruction that gets the function pointer of an external function from its name and an address.

### CODE REVIEW

File: tests/iele/unit/fptr.iele [hunk 1]
```diff
@@ -27,6 +27,8 @@ contract test {
     revert -1
   }
 
+  @unused = 13
+
```
Comment (theo25) [inline tests/iele/unit/fptr.iele]: Why was that added?

Comment (dwightguth) [reply tests/iele/unit/fptr.iele]: Testing that we still compute the right offset for test2

File: compiler/src/IelePrint.hs [hunk 2]
```diff
@@ -162,6 +162,13 @@ prettyIeleInst (CallOp (STATICCALLDYN _ _) results allArgs) = case results of
              <> char '(' <> commaList args <> char ')'
              <+> text "gaslimit" <+> gas
          _ -> error "external staticcall instruction must encode at least target and gaslimit arguments"
+prettyIeleInst (CallOp (CALLADDRESS name) results allArgs) = case results of
+  [] -> error "external call instruction must have at least one result"
+  _ -> case allArgs of
+         (acct:[]) ->
+           prettyResults results <+> text "calladdress" <+> prettyName name
+             <+> text "at" <+> acct
+         _ -> error "external staticcall instruction must encode at least target argument"
```
Comment (bmmoore) [inline compiler/src/IelePrint.hs]: This message says "staticcall". This should also expect exactly one result, right?

File: compiler/src/IeleInstructions.hs [hunk 3]
```diff
@@ -205,6 +206,7 @@ retypeOpcodeCall contract fun i = case i of
     STATICCALLDYN nargs nrets -> pure (STATICCALLDYN nargs nrets)
     LOCALCALL f nargs nrets -> (\f -> LOCALCALL f nargs nrets) <$> fun f
     LOCALCALLDYN nargs nrets -> pure (LOCALCALLDYN nargs nrets)
+    CALLADDRESS f -> (\f -> CALLADDRESS f) <$> fun f
```
Comment (bmmoore) [inline compiler/src/IeleInstructions.hs]: This could be `CALLADDRESS <$> fun f`
